### PR TITLE
Client - Update Get Message Metadata

### LINKED ISSUE #299: Client - update get_message_meta to v2
## Description

The message meta endpoint has been changed
- update restful API route to v2
- update the response format
- `test_message_meta` tests should be updated and passed

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

Fixed #299 

## Type of change

- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

New unit tests are in `test_message_meta.c` file.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/client/api_restful/test_message_meta.c [hunk 1]
```diff
@@ -12,79 +12,123 @@ void setUp(void) {}
 
 void tearDown(void) {}
 
-void test_deser_mileston_message() {
-  char const* const json_mileston =
-      "{\"data\":{\"messageId\":\"ebe0c79284d318a1bc36f1d2b6a94ea2ada27a1334d40c565823040131aa1ab4\","
+void test_deser_milestone_message() {
```
Comment (oopsmonk) [inline tests/client/api_restful/test_message_meta.c]:
```suggestion
void test_deser_milestone_meta() {
```

File: tests/client/api_restful/test_message_meta.c [hunk 2]
```diff
@@ -12,79 +12,123 @@ void setUp(void) {}
...
   TEST_ASSERT(meta->u.meta->referenced_milestone == 285132);
   TEST_ASSERT(meta->u.meta->milestone_idx == 285132);
   TEST_ASSERT_EQUAL_STRING("noTransaction", meta->u.meta->inclusion_state);
   TEST_ASSERT(meta->u.meta->should_promote == -1);
   TEST_ASSERT(meta->u.meta->should_reattach == -1);
-  res_msg_meta_free(meta);
+  msg_meta_free(meta);
 }
 
 void test_deser_tx_message() {
```
Comment (oopsmonk) [inline tests/client/api_restful/test_message_meta.c]:
```suggestion
void test_deser_tx_meta() {
```

File: tests/client/api_restful/test_message_meta.c [hunk 3]
```diff
@@ -12,79 +12,123 @@ void setUp(void) {}
...
   TEST_ASSERT(meta->u.meta->referenced_milestone == 161055);
   TEST_ASSERT(meta->u.meta->milestone_idx == 0);
   TEST_ASSERT_EQUAL_STRING("included", meta->u.meta->inclusion_state);
   TEST_ASSERT(meta->u.meta->should_promote == -1);
   TEST_ASSERT(meta->u.meta->should_reattach == -1);
-  res_msg_meta_free(meta);
+  msg_meta_free(meta);
+}
+
+void test_deser_tagged_data_message() {
```
Comment (oopsmonk) [inline tests/client/api_restful/test_message_meta.c]:
```suggestion
void test_deser_tagged_data_meta() {
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]
function names with `message` are confusing, please replace them with `meta`.

File: tests/client/api_restful/test_message_meta.c (hunk 1 above)
Comment (matjazv) [reply tests/client/api_restful/test_message_meta.c]: Function name was changed.

File: tests/client/api_restful/test_message_meta.c (hunk 2 above)
Comment (matjazv) [reply tests/client/api_restful/test_message_meta.c]: Function name was changed.

File: tests/client/api_restful/test_message_meta.c (hunk 3 above)
Comment (matjazv) [reply tests/client/api_restful/test_message_meta.c]: Function name was changed.

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**